kedro/config/omegaconf_config.py

```diff
@@ -109,6 +110,7 @@ def __init__(  # noqa: too-many-arguments
             "parameters": ["parameters*", "parameters*/**", "**/parameters*"],
             "credentials": ["credentials*", "credentials*/**", "**/credentials*"],
             "logging": ["logging*", "logging*/**", "**/logging*"],
+            "globals": ["globals*", "globals*/**", "**/globals*"],
```

Comment:
```
        >>> # in settings.py
        >>> from kedro.config import TemplatedConfigLoader
        >>>
        >>> CONFIG_LOADER_CLASS = TemplatedConfigLoader
        >>> CONFIG_LOADER_ARGS = {
        >>>     "globals_pattern": "*globals.yml",
        >>> }
```
```suggestion
            "globals": ["globals*"],
```

I suggest keep this pattern simple.
1. Align with the current `TemplatedConfigLoader` pattern, the above code block is copy from our template
2. Most people don't need nested `globals`,  putting globals in folder should be minority, if so they should change the settings instead.
3. `**/globals*` is quite dangerous, as this could match `parameters/globals_parameters.yml` - although we usually suggest people using `parameter_globals` instead.

I may even suggest a more conservative default  `"globals": "globals.yml"`, forcing the default to be just `globals.yml`. If we change this after release it will be breaking change.

WDYT? @ankatiyar @merelcht , cc @stichbury because I really think we should separate the terminology for `globals`, it's being overloaded a lot now and this can lead to weird bugs.

Reply:
Offline conversation - agree with using default with `globals.yml`.